server/response/apiresponse.go

```diff
@@ -0,0 +1,44 @@
+package response
+
+type ApiResponse struct {
+	Code int
+	Body ApiResponseBody
+}
+
+type ApiResponseBody struct {
+	Data    interface{} `json:"data,omitempty"`
+	Error   interface{} `json:"error,omitempty"`
+}
+
+type BaseMessage struct {
+	Message string `json:"message"`
+}
+
+type ApiError struct {
+	BaseMessage
+	Data interface{} `json:"data,omitempty"`
+}
+
+func NewOkResponse(data interface{}) ApiResponse {
+	return ApiResponse{
+		Body: ApiResponseBody{
+			Data: data,
+		},
+	}
+}
+
+func NewErrorResponse(code int, message BaseMessage, data interface{}) ApiResponse {
```

Comment:
removing the BaseMessage stuff simplifies this because then message can just be a string. i would also remove the data arg in favor of adding another func called NewErrorResponseWithData for cases where error data is needed. that way the errors that dont have data won't have a bunch of `nil` args in them.